packages/matchbox/src/components/ProgressBar/ProgressBar.js

```diff
@@ -1,45 +1,51 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
-import classnames from 'classnames';
-import styles from './ProgressBar.module.scss';
-
-class ProgressBar extends Component {
-  static displayName = 'ProgressBar';
-
-  static propTypes = {
-    /**
-     * Completion in percentage
-     */
-    completed: PropTypes.number.isRequired,
-
-    /**
-     * Bar color
-     */
-    color: PropTypes.oneOf(['orange', 'blue', 'navy', 'purple', 'red'])
-  };
-
-  static defaultProps = {
-    completed: 0,
-    color: 'orange'
-  }
-
-  render() {
-    const { completed = 0, color } = this.props;
-
-    let percentage = completed;
-
-    if (percentage > 100) {
-      percentage = 100;
-    } else if (percentage < 1) {
-      percentage = 0;
-    }
-
-    return (
-      <div className={classnames(styles.ProgressBar, styles[color])}>
-        <div className={styles.Progress} style={{ width: `${percentage}%` }}/>
-      </div>
-    );
-  }
+import { deprecate } from '../../helpers/propTypes';
+import styled from 'styled-components';
+import { margin, compose } from 'styled-system';
+import { createPropTypes } from '@styled-system/prop-types';
+import { Box } from '../Box';
+
+import { outerBase, innerBase, visualSize, calculatedWidth } from './styles.js';
+
+const system = compose(margin);
+
+export const StyledProgressBarOuter = styled(Box)`
+  ${outerBase}
+  ${visualSize}
+  ${system}
+`;
+
+export const StyledProgressBarInner = styled(Box)`
+  ${innerBase}
+  ${visualSize}
+  ${calculatedWidth}
+`;
+
+function ProgressBar(props) {
+  const { completed = 0, size } = props;
+
+  return (
+    <StyledProgressBarOuter as="div" size={size}>
```

Comment:
Will need to spread props here to get any of the margin props passed through 
```js
const { completed = 0, size, ...rest } = props;
<StyledProgressBarOuter as="div" size={size} {...rest}>
```

Reply:
Spreading props is a bad pattern though that we'll have to fix eventually but I'll create an issue to fix this sort of thing holistically, at least for system props